first_app/hello.py

```diff
@@ -0,0 +1,184 @@
+    return {"info": user.user_info()},200
+@app.route("/api/v1/user-info/edit/<int:user_id>", methods=['PUT'])
+@login_required
+def user_profile_edit(user_id):
+    if user_id != g.user_id:
+        return{"error": "Request data isn't yours"}
+    data = request.json
+    new_phone_number = data["phone_number"]
+    new_first_name = data["first_name"]
+    new_second_name = data["second_name"]
+    exists = db_session.query(User.query.filter(User.phone_number==new_phone_number).exists()).scalar()
+    if exists:
+        return{"error:":"Phone number already exists"}, 400
+    User.query.filter(User.id==user_id).update({"phone_number":new_phone_number,
+                                                "first_name": new_first_name,
+                                                "second_name":new_second_name},
+                                                synchronize_session=False)
+    db_session.commit()
+    return{"return":"ok"}
+@app.route("/api/v1/user-info/change-password/<int:user_id>", methods=['PUT'])
+@login_required
+def change_password(user_id):
+    if user_id != g.user_id:
+        return{"error": "Request data isn't yours"}
+    data = request.json
+    old_password = data["old_password"]
+    new_password = data["new_password"]
+    user = User.query.filter(User.id==user_id).one()
+    if not new_password:
+        return{"error":"Немає нового паролю"}
+    if not check_password_hash(user.password,old_password):
+        return{"error":"pa"}    
+    User.query.filter(User.id==user_id).update({"password":generate_password_hash(new_password)}) 
+    db_session.commit()
+    return{}
+@app.route("/api/v1/create-post/<int:user_id>", methods=["POST"])
+@login_required
+def create_post_api(user_id):
+    db = get_db()
+    data = request.json
+    if user_id == g.user_id:
+        return{"error":"ви пробуєте створити не у себе пост"}
+    author_id = user_id
+    title = data['title']
+    body = data['body']
+    current_datetime = datetime.datetime.today() 
+    if not title or body:    
+        return {"error": "немає title, або body"}
+    else:
+        new_post = Post(author_id,current_datetime,title,body)
+        db_session.add(new_post)
+        db_session.commit()
+    
+    return "Опубліковано", 200
+@app.route("/api/v1/user-posts/<int:user_id>", methods=["GET"])
+@login_required
+def user_post_api(user_id):
+    db = get_db()
+    income_author_id = user_id
+    posts_db = Post.query.filter(Post.author_id==income_author_id and Post.deleted==0).all()
+    posts = []
+    for post in posts_db:
+        posts.append({
+            "author id": post.author_id,
+            "title": post.title,
+            "body": post.body,
+            "created": post.created
+        })
+    return posts, 200
+@app.route("/api/v1/delete-post/<int:user_id>/delete/<int:post_id>", methods=["DELETE"])
+@login_required
+def delet_post_api(user_id, post_id):
+    data = request.json  
+    if user_id == g.user_id:
```

Comment:
`
post = Post.query.filter(Post.id == post_id, Post.author_id == user_id).first()
if not post:
    return {"error": "Post not found"}
post.deleted = 1
db.session.add(post)
db.session.commit()
`